src/ui/hook/useCopyToClipboard.ts

```diff
@@ -0,0 +1,36 @@
+import { useCallback, useState } from 'react'
+import { useTimeout } from '@/ui/hook/useTimeout'
+
+type HandleCopy = (textToCopy: string) => Promise<void>
+type CopyState = 'idle' | 'success' | 'error'
```

Comment:
Please go further in the definition of the algebraic type, by making the copied text for the `success` state, and the error, for the `error` state.